app/src/main/java/nu/yona/app/ui/YonaActivity.java

```diff
@@ -428,6 +406,44 @@ public void onClick(DialogInterface dialogInterface, int i)
 		}
 	}
 
+	private void showUserNotFoundAlertDialog(ErrorMessage errorMessage)
+	{
+		try
+		{
+			CustomAlertDialog.show(YonaActivity.this, errorMessage.getMessage(), getString(R.string.ok), new DialogInterface.OnClickListener()
+			{
+				@Override
+				public void onClick(DialogInterface dialogInterface, int i)
+				{
+					updateSharedPreferences();
+					dialogInterface.dismiss();
+					clearDatabaseAndNavigateToLaunchActivity();
```

Comment:
Why are you clearing the database after the dismiss? I'd expect that to happen together with updating (resetting) the shared preferences. That's also in line with the original code.

I propose to create two methods:
1) ``resetData()``, resetting the preferences and the database
2) ``navigateToLaunchActivity``, as navigate to launch activity.